### PR TITLE
[v0.14] Cherry-pick Inference Graph stabilizations

### PR DESCRIPTION
**Pull request by israel-hdez**
**What this PR does / why we need it**:

Cherry-pick the following PRs:
* kserve/kserve#4031
* opendatahub-io/kserve#426

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://issues.redhat.com/browse/RHOAIENG-13448

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: pkg/controller/v1alpha1/inferencegraph/raw_ig.go [hunk 1]
```diff
@@ -148,7 +148,7 @@ func handleInferenceGraphRawDeployment(cl client.Client, clientset kubernetes.In
 	reconciler, err := raw.NewRawKubeReconciler(cl, clientset, scheme, objectMeta, &componentExtSpec, desiredSvc)
 
 	if err != nil {
-		return nil, reconciler.URL, errors.Wrapf(err, "fails to create NewRawKubeReconciler for inference graph")
+		return nil, nil, errors.Wrapf(err, "fails to create NewRawKubeReconciler for inference graph")
```
Comment (spolti) [inline pkg/controller/v1alpha1/inferencegraph/raw_ig.go:151]: only one question, do you know why is this needed?

Comment (israel-hdez) [reply pkg/controller/v1alpha1/inferencegraph/raw_ig.go:151]: This is actually a bugfix cherry-picked from the [upstream PR](https://github.com/kserve/kserve/pull/4031).

In CI sometimes it gives a panic. Since `err` is _not_ nil, the `reconciler` variable is nil.

**Review by spolti — APPROVED** [review]
